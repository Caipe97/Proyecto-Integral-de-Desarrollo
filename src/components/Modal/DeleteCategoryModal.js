import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import Combobox from "react-widgets/Combobox";
import "react-widgets/styles.css";
import Delete from '../../images/delete.png';

const DeleteCategoryModal = (props) =>{
  const [modalShow, setModalShow] = useState(false);
  let onlyCustomCategories = props.foodCategories.filter((category) => category.userId !== null);

  const [state, setState] = useState({
    name: '',
    foodCategoryId: '',
    categoryName: '',
    message: ''
  })

  const handleSubmit = async (event) => {
    event.preventDefault();
    if(state.foodCategoryId !== ''){
      const data = await props.onDeleteCategory(state.foodCategoryId);
      if(data){
        setState({
          name: '',
          foodCategoryId: '',
          categoryName: '',
          message: 'Categoría eliminada exitosamente'
        });
      }
    } else {
      setState({...state, message: 'Debe seleccionar alguna categoría para eliminarla'});
    }
  };

  const getFoodCategoryName = (foodCategoryId) => {
    const foodCategoryName = props.foodCategories.find(foodCategory => foodCategory.foodCategoryId === foodCategoryId);
    if(foodCategoryId === -2){
      return '';
    }
    return foodCategoryName;
  }

  const onChangeComboBox = async event => {
    await setState({...state, foodCategoryId: event.foodCategoryId, categoryName: getFoodCategoryName(event.foodCategoryId)});
  }

  return (
    <>
      <img src={Delete} alt='eliminar' style={{width: '30px', height: '30px', cursor: 'pointer'}} onClick={() => setModalShow(true)}/>
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {'Eliminar Categoría'}
          </Modal.Title>
          <Button type="button" onClick={() => {setModalShow(false); setState({...state, message: ''});}} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
        </Modal.Header>
        <Modal.Body>

          <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
            <Combobox
              data={onlyCustomCategories}
              value={state.categoryName}
              textField='name'
              onSelect={onChangeComboBox}
              groupBy={category => category.userId}
              renderListGroup={ ({group}) => ( //group es el userId
                <span>{group === null ? 'Default' : 'Custom'}</span>
              )}
            />
            <button onClick={handleSubmit} className='button'>Eliminar</button>
            <p>{state.message}</p>
            {props.isPending
              ? <div className="spinner-border" role="status">
                  <span className="sr-only"></span>
                </div>
              : null}
          </form>
        </Modal.Body>
      </Modal>
    </>
  );
}
  
  export default DeleteCategoryModal;