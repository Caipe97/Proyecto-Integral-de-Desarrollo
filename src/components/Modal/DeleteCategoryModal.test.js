import React from 'react';
import { shallow } from 'enzyme';
import DeleteCategoryModal from './DeleteCategoryModal';

describe('ModalDeleteCategoryModal tests', () => {
    const preventDefault = { preventDefault: jest.fn() }
    let wrapper;
    beforeEach(() => {
        let mockProps = {
            state: {
                name: '',
                foodCategoryId: '',
                errorMessage: ''
            },
            foodCategories:[
                {
                  foodCategoryId: 1,
                  name: "Fruta",
                  createdAt: "2021-09-30T14:09:34.756Z",
                  updatedAt: "2021-09-30T14:09:34.756Z",
                  userId: null
                },
                {
                  foodCategoryId: 7,
                  name: "Fideos",
                  createdAt: "2021-09-30T18:58:58.542Z",
                  updatedAt: "2021-09-30T18:58:58.542Z",
                  userId: null
                }
              ],
            preventDefault: jest.fn(),
            onDeleteCategory: jest.fn()
        };
        wrapper = shallow(<DeleteCategoryModal {...mockProps} />

        );

    })
    it('expect to render DeleteCategoryModal component to render when edit is false', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('Set modal and setstate null message expect to change  onclick buttons', () => {

        wrapper.find('[type="button"]').at(0).simulate('click');
        expect(wrapper.find('div')).toBeDefined();
        expect(wrapper.find('div.p').contains('')).toBeDefined();

    })
    it('expect to change  onclick buttons for set modal', () => {

        wrapper.find('img').at(0).simulate('click');

    })
    it('handleSubmit  expect to change onclick for delete',  () => {
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);

    })

})
