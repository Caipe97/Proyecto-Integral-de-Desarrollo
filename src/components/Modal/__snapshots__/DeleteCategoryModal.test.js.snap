// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDeleteCategoryModal tests expect to render DeleteCategoryModal component to render when edit is false 1`] = `
<Fragment>
  <img
    alt="eliminar"
    onClick={[Function]}
    src="delete.png"
    style={
      Object {
        "cursor": "pointer",
        "height": "30px",
        "width": "30px",
      }
    }
  />
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    restoreFocus={true}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        Eliminar Categor√≠a
      </ModalTitle>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "color": "black",
          }
        }
        type="button"
        variant="primary"
      >
        X
      </Button>
    </ModalHeader>
    <ModalBody>
      <form
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "center",
          }
        }
      >
        <Combobox
          data={Array []}
          groupBy={[Function]}
          onSelect={[Function]}
          renderListGroup={[Function]}
          textField="name"
          value=""
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Eliminar
        </button>
        <p />
      </form>
    </ModalBody>
  </Modal>
</Fragment>
`;
